nifi-extension-bundles/nifi-poi-bundle/nifi-poi-services/src/main/java/org/apache/nifi/excel/ExcelRecordReader.java

```diff
@@ -127,17 +134,6 @@ private Map<String, Object> getCurrentRowValues(Row currentRow, boolean coerceTy
         return currentRowValues;
     }
 
-    private static Object getCellValue(Cell cell) {
-        if (cell != null) {
-            return switch (cell.getCellType()) {
-                case _NONE, BLANK, ERROR, FORMULA, STRING -> cell.getStringCellValue();
-                case NUMERIC -> DateUtil.isCellDateFormatted(cell) ? cell.getDateCellValue() : cell.getNumericCellValue();
-                case BOOLEAN -> cell.getBooleanCellValue();
```

Comment:
Does the `DataFormatter` sufficiently handle all of these cases now?

Reply:
I believe it does but I will add a unit test to demonstrate it does.

Reply:
@exceptionfactory I am glad you brought up this point since as a result I discovered there was bug in formula results as they were not being displayed rather the actual formula was being retrieved. I have fixed that with the latest changes and I have expanded the ExcelReader unit tests to exercise all types.

Reply:
@exceptionfactory I see the build is failing on the Windows box which has the French Locale and in France the separator is a comma and not a period. The stacktrace I see is

```
Caused by: java.lang.NumberFormatException: For input string: "18,36"
	at java.base/jdk.internal.math.FloatingDecimal.readJavaFormatString(FloatingDecimal.java:2054)
	at java.base/jdk.internal.math.FloatingDecimal.parseFloat(FloatingDecimal.java:122)
	at java.base/java.lang.Float.parseFloat(Float.java:556)
	at org.apache.nifi.serialization.record.util.DataTypeUtils.toFloat(DataTypeUtils.java:1330)
	at org.apache.nifi.serialization.record.util.DataTypeUtils.convertType(DataTypeUtils.java:209)
	at org.apache.nifi.serialization.record.util.DataTypeUtils.convertType(DataTypeUtils.java:174)
	at org.apache.nifi.excel.ExcelRecordReader.convert(ExcelRecordReader.java:134)
	at org.apache.nifi.excel.ExcelRecordReader.lambda$getCurrentRowValues$0(ExcelRecordReader.java:119)
	at java.base/java.util.stream.Streams$RangeIntSpliterator.forEachRemaining(Streams.java:104)
	at java.base/java.util.stream.IntPipeline$Head.forEach(IntPipeline.java:617)
	at org.apache.nifi.excel.ExcelRecordReader.getCurrentRowValues(ExcelRecordReader.java:106)
```

Can this be a bug in `DataTypeUtils` which does not take into account the `Locale` when parsing a string to a number?

Reply:
I think the strategy of using the `DataFormatter` in all cases is not great because of the string localization. The `getNumericValue()` method seems like a better option if that works consistently.

Reply:
The problem I see is I have to eventually pass it as a string to `SchemaInferenceUtil.getDataType` . I have to ensure the `toString` method of `Double` does not take the locale into account.

Reply:
Seems like it wasn't an issue as the build passed.